### PR TITLE
Move dcStream to new project Deflect

### PR DESCRIPTION
**Pull request by tribal-tec**
Change-Id: I43c633a94f11e6ed8b9ca358de044ebbac7aa67a

### CODE REVIEW

File: apps/LocalStreamer/CMakeLists.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@ set(LOCALSTREAMER_SOURCES
   src/Application.cpp
 )
 
-set(LOCALSTREAMER_LINK_LIBRARIES dccore dcstream)
+set(LOCALSTREAMER_LINK_LIBRARIES dccore)
```
Comment (rdumusc) [inline apps/LocalStreamer/CMakeLists.txt]: No need to link with Deflect here?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -11,23 +11,7 @@ else()
...
-  ${CMAKE_CURRENT_SOURCE_DIR}/PixelStreamSegmentParameters.h
-)
-
-set(DISPLAYCLUSTER_SHARED_SOURCES
-  ${CMAKE_CURRENT_SOURCE_DIR}/Event.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/log.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/MessageHeader.cpp
-)
-
 if(BUILD_CORE_LIBRARY)
```
Comment (rdumusc) [inline src/CMakeLists.txt]: We should do more changes here, CORE_LIBRARY should always be build now that the stream library is separated

File: src/core/MPIChannel.cpp [hunk 3]
```diff
@@ -40,7 +40,7 @@
 #include "MPIChannel.h"
 
 #include "MPIContext.h"
-#include "MessageHeader.h"
```
Comment (rdumusc) [inline src/core/MPIChannel.cpp]: Oh, this include should already have been removed! There is a different MPIHeader.h included in MPIChannel.h

File: src/core/MetaTypeRegistration.cpp [hunk 4]
```diff
@@ -55,7 +55,7 @@ struct MetaTypeRegistration
     MetaTypeRegistration()
     {
         qRegisterMetaType< size_t >( "size_t" );
-        qRegisterMetaType< Event >( "Event" );
+        qRegisterMetaType< deflect::Event >( "Event" );
```
Comment (rdumusc) [inline src/core/MetaTypeRegistration.cpp]: There is some inconsistency that we use the fully qualifying deflect::Event but only PixelStreamSegment. Not sure which one to settle for?

File: src/core/CMakeLists.txt [hunk 5]
```diff
@@ -72,6 +73,7 @@ list(APPEND DCCORE_PUBLIC_HEADERS
   gestures/PinchGesture.h
   gestures/PinchGestureRecognizer.h
   ImageJpegDecompressor.h
+  ../log.h
```
Comment (rdumusc) [inline src/core/CMakeLists.txt]: Last thing, can we move log.* to the core folder and get rid of "include_directories(src)" in the top-level CMakeLists?

File: apps/LocalStreamer/CMakeLists.txt (hunk 1 above)
Comment (tribal-tec) [inline apps/LocalStreamer/CMakeLists.txt]: Works without at least on Ubuntu, but I'll put it back

File: src/core/CMakeLists.txt (hunk 5 above)
Comment (tribal-tec) [inline src/core/CMakeLists.txt]: The include_directories(src) is still necessary due to inconsistencies of the folder layout in the source folder vs install folder. The dcWebservice inclusion in the core library is the point where it is still needed.

**Comment by tribal-tec** [comment]
Retest please

**Comment by tribal-tec** [comment]
Retest this please

**Comment by rdumusc** [comment]
+1

**Comment by tribal-tec** [comment]
Retest this please

**Pull request merged**